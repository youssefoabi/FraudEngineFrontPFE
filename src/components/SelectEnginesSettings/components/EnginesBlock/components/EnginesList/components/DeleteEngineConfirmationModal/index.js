import { bool, func, string } from 'prop-types';
import { isNilOrEmpty } from 'ramda-adjunct';
import { Typography } from '@material-ui/core';
import React from 'react';

import ConfirmationModal from '../../../../../../../ConfirmationModal';

export default function DeleteEngineConfirmationModal({ engineName, confirm, open, toggle }) {
  if (isNilOrEmpty(engineName)) return null;

  return (
    <ConfirmationModal isOpen={open} title="Suppression du moteur" toggle={toggle} confirm={confirm}>
      <Typography>{`Confirmez-vous la suppression du moteur  ${engineName} ?`}</Typography>
    </ConfirmationModal>
  );
}

DeleteEngineConfirmationModal.propTypes = {
  open: bool,
  toggle: func,
  engineName: string,
  confirm: func,
};

DeleteEngineConfirmationModal.defaultProps = {
  open: false,
  toggle: Function.prototype,
  engineName: null,
  confirm: Function.prototype,
};
