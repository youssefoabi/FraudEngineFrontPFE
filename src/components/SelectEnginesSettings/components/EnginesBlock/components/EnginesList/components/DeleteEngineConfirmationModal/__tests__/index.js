import { itRendersAllMutations } from '@cdiscount/jest-utils';

import DeleteEngineConfirmationModal from '../';

const mutations = [
  {
    name: 'with engines',
    props: { engineName: 'engine', confirm: jest.fn(), open: true, toggle: jest.fn() },
  },
  { name: 'without props' },
];

describe(DeleteEngineConfirmationModal.name, () => {
  itRendersAllMutations(DeleteEngineConfirmationModal, mutations);
});
