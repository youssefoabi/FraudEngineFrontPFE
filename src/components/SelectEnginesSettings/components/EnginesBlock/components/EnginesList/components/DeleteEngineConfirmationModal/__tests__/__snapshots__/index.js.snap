// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteEngineConfirmationModal DeleteEngineConfirmationModal defaultProp functions runs defaultProps.confirm() 1`] = `undefined`;

exports[`DeleteEngineConfirmationModal DeleteEngineConfirmationModal defaultProp functions runs defaultProps.toggle() 1`] = `undefined`;

exports[`DeleteEngineConfirmationModal DeleteEngineConfirmationModal renders correctly with engines 1`] = `
<ConfirmationModal
  confirm={[MockFunction]}
  isOpen={true}
  title="Suppression du moteur"
  toggle={[MockFunction]}
>
  <WithStyles(ForwardRef(Typography))>
    Confirmez-vous la suppression du moteur  engine ?
  </WithStyles(ForwardRef(Typography))>
</ConfirmationModal>
`;

exports[`DeleteEngineConfirmationModal DeleteEngineConfirmationModal renders correctly without props 1`] = `""`;
